test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs

```diff
@@ -20,83 +17,108 @@ public static class TestHelper
 
+        private static IAssemblyProvider CreateAssemblyProvider(string siteName)
+        {
+            // Creates a service type that will limit MVC to only the controllers in the test site.
+            // We only want this to happen when running in-process.
+            var assembly = Assembly.Load(new AssemblyName(siteName));
+            var provider = new FixedSetAssemblyProvider
+            {
+                CandidateAssemblies =
+                {
+                    assembly,
+                },
+            };
+
+            return provider;
+        }
+
         /// <summary>
-        /// Creates a disposable action that replaces the service provider <see cref="CallContextServiceLocator.Locator"/>
-        /// with the passed in service that is switched back on <see cref="IDisposable.Dispose"/>.
+        /// Creates a disposable action that replaces the service provider at
+        /// <see cref="CallContextServiceLocator.Locator"/> with one that overrides a couple of host services.
+        /// Service provider is switched back on <see cref="IDisposable.Dispose"/>.
         /// </summary>
-        /// <remarks>This is required for config since it uses the static property to get to
-        /// <see cref="IApplicationEnvironment"/>.</remarks>
-        public static IDisposable ReplaceCallContextServiceLocationService(IServiceProvider serviceProvider)
+        /// <remarks>
+        /// This is required for config since it uses the static property to get to
+        /// <see cref="IApplicationEnvironment"/>.
+        /// </remarks>
+        public static IDisposable ReplaceCallContextServiceLocationService(string applicationWebSiteName)
         {
-            return new CallContextProviderAction(serviceProvider);
+            return ReplaceCallContextServiceLocationService(applicationWebSiteName, WebsitesDirectoryPath);
         }
 
-        private static Type CreateAssemblyProviderType(string siteName)
+        /// <summary>
+        /// Creates a disposable action that replaces the service provider at
+        /// <see cref="CallContextServiceLocator.Locator"/> with one that overrides a couple of host services.
+        /// Service provider is switched back on <see cref="IDisposable.Dispose"/>.
+        /// </summary>
+        /// <remarks>
+        /// This is required for config since it uses the static property to get to
+        /// <see cref="IApplicationEnvironment"/>.
+        /// </remarks>
+        public static IDisposable ReplaceCallContextServiceLocationService(
+            string applicationWebSiteName,
+            string applicationPath)
         {
-            // Creates a service type that will limit MVC to only the controllers in the test site.
-            // We only want this to happen when running in proc.
-            var assembly = Assembly.Load(new AssemblyName(siteName));
+            var serviceCollection = HostingServices.Create(
+                CallContextServiceLocator.Locator.ServiceProvider,
```

Comment:
I added a new Create overload today that lets you omit the CallContextServiceLocator:

https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNet.Hosting/HostingServices.cs#L46